File: C/ABC/ABC270B.c
#include <stdio.h>
#include <stdlib.h>

int main(void)
{
    int X, Y, Z;
    if (scanf("%d %d %d", &X, &Y, &Z) != 3)
        return 0;

    if ((0 < Y && Y < X && X < Z) ||
        (0 < Y && Y < Z && Z < X) ||
        (X < Z && Z < Y && Y < 0) ||
        (Z < X && X < Y && Y < 0))
    {
        printf("-1\n");
    }
    else if ((Z < 0 && 0 < Y && Y < X) ||
             (X < Y && Y < 0 && 0 < Z))
    {
        printf("%d\n", 2 * abs(Z) + abs(X));
    }
    else
    {
        printf("%d\n", abs(X));
    }

    return 0;
}

File: python/ABC/ABC270B.py
X, Y, Z = map(int, input().split())

# 座標4つの並びは4P4だが、左右対処なので、Y<0に固定すると4P4の半分で12通り
if Y < 0:
    X, Y, Z = -X, -Y, -Z

if X < Y:  # ゴールが壁の手前
    print(abs(X))
elif Y < Z:  # ハンマーが壁の奥
    print(-1)
else:  # ゴールと反対方向にハンマーを拾いに行ってからゴール
    print(abs(Z) + abs(X - Z))
